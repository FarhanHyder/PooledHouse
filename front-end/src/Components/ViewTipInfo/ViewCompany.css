.company{
    position: fixed;
    width: 100vw;
}
.companyProfile {
    display: grid;
    width: 100vw;
    height: 500px;
    overflow:auto;
    background:#fff;
    position: fixed;
}

/* .menu {
    position: fixed;
    padding: 10px 10px 10px 10px;
} */

/* .Search {
    display: grid;
    grid-auto-flow: column;
    width: 500px;
} */
