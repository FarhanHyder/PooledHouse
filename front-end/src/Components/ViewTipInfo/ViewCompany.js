/**
 * @class ViewCompany
 * @extends Component
 */
import React, { Component } from 'react';
import Description from './CompanyDescription';
import ViewReportedTips from './ViewReportedTips';
import ViewTipsHistory from './ViewTipsHistory';
const process = require('../ProcessTips/ProcessTips');

class ViewCompany extends Component {
    constructor(props){
        super(props);
        this.state = {
            BusinessName: this.props.BusinessName,
            allBusiness: process.createBusinessTable(this.props.tipsInfo),
            avgByLocation: process.averageTipsByBusinessByLocation(this.props.tipsInfo),
            avgByPosition: process.averageTipsByPosition(this.props.tipsInfo),
            avgByDay: process.averageTipsByBusinessDay(this.props.tipsInfo),
            avgByZip: process.averageTipsByZipCode(this.props.tipsInfo),
            avgByPosShift: process.averageTipsByPositionAndShift(this.props.tipsInfo),
            ViewDescription: true,
            ViewTipsInfo: false,
            ViewTipsHistory: false,
        }
        this.companyViewHandler = this.companyViewHandler.bind(this);
        this.descriptionViewHandler = this.descriptionViewHandler.bind(this);
        this.tipsInfoViewHandler = this.tipsInfoViewHandler.bind(this);
        this.tipsHistoryViewHandler = this.tipsHistoryViewHandler.bind(this);
    }

    companyViewHandler = () => {
        this.setState({CompanyView : !this.state.CompanyView});
    };

    descriptionViewHandler = () => {
        this.setState({
            ViewDescription: true,
            ViewTipsInfo: false,
            ViewTipsHistory: false
        });
    };
    tipsInfoViewHandler = () => {
        this.setState({
            ViewDescription: false,
            ViewTipsInfo: true,
            ViewTipsHistory: false
        });
    };
    tipsHistoryViewHandler = () => {
        this.setState({
            ViewDescription: false,
            ViewTipsInfo: false,
            ViewTipsHistory: true
        });
    };

    render(){
        const Company = (
            <div className = "container">
                {/* <div className="card"> */}
                <div className="position-left">
                    <h2 className="card-title text-white bg-success rounded-lg">{this.state.BusinessName}</h2>
                    <div className = "row">
                        <div className="col-6 btn-group">
                            <button type="button" className="btn btn-info" onClick = {this.descriptionViewHandler}>Description</button>
                            <button type="button" className="btn btn-dark" onClick = {this.tipsInfoViewHandler}>Tips Information</button>
                            <button type="button" className="btn btn-secondary" onClick = {this.tipsHistoryViewHandler}>Tips History</button>
                        </div>
                        <h5 className=" text-right col-6 text-success position-right">Average Tips ${Number.parseFloat(this.props.avgByBusiness.tipsPerHour.toFixed(2))}</h5>
                    </div>
                </div>
               {/* </div> */}
                <div className="row">
                    {
                        this.state.ViewDescription?
                        <Description
                            business = {this.state.BusinessName}
                            locations = {this.state.avgByLocation[this.state.BusinessName]}
                            avgByZip = {this.state.avgByZip}
                        /> :
                        (this.state.ViewTipsInfo?
                        <ViewReportedTips
                            business = {this.state.BusinessName}
                            locations = {this.state.avgByLocation[this.state.BusinessName]}
                            avgByDay = {this.state.avgByDay[this.state.BusinessName]}
                            avgByPosition = {this.state.avgByPosition[this.state.BusinessName]}
                            avgByPosShift = {this.state.avgByPosShift[this.state.BusinessName]}
                        /> : 
                        
                        <ViewTipsHistory
                            tipsHistory = {this.state.allBusiness[this.state.BusinessName]}
                        />)
                    }
                </div>
            </div>
        );
        
        return Company;
    }
}

export default ViewCompany;
