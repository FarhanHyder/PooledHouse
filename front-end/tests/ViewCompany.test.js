import React from 'react';
import renderer from 'react-test-renderer';
import ViewCompany from '../src/Components/ViewTipInfo/ViewCompany';

test('ViewCompany component renders correct for single element', () => {
  const thisProps = {
    BusinessName: 'Poolinos',
    tipsInfo: [
    {
      'shift_length': 5,
      'shift_time': 'PM',
      'shift_date': '04/30/2019',
      'shift_position': 'Busser',
      'takehome': 87,
      'business_name': 'Poolinos',
      'business_street_address': '292 Bowery',
      'business_city': 'New York',
      'business_state': 'NY',
      'business_zip': '10012',
      'neighborhood': 'Soho'
    }],
    avgByBusiness: {
      tipsPerHour: 12.8888
    }
  };

  const component = renderer.create(
//    <ViewCompany props={thisProps} />
    <ViewCompany BusinessName='Poolinos' tipsInfo={thisProps.tipsInfo} avgByBusiness={thisProps.avgByBusiness} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test('ViewCompany component renders correct for multiple elements', () => {
  const thisProps = {
    BusinessName: 'Poolinos',
    tipsInfo: [
    {
      'shift_length': 5,
      'shift_time': 'PM',
      'shift_date': '04/30/2019',
      'shift_position': 'Busser',
      'takehome': 87,
      'business_name': 'Poolinos',
      'business_street_address': '292 Bowery',
      'business_city': 'New York',
      'business_state': 'NY',
      'business_zip': '10012',
      'neighborhood': 'Soho'
    },
    {
      'shift_length': 6,
      'shift_time': 'PM',
      'shift_date': '05/02/2019',
      'shift_position': 'Busser',
      'takehome': 93,
      'business_name': 'Poolinos',
      'business_street_address': '292 Bowery',
      'business_city': 'New York',
      'business_state': 'NY',
      'business_zip': '10012',
      'neighborhood': 'Soho'
    }],
    avgByBusiness: {
      tipsPerHour: 12.8888
    }
  };

  const component = renderer.create(
    <ViewCompany BusinessName='Poolinos' tipsInfo={thisProps.tipsInfo} avgByBusiness={thisProps.avgByBusiness} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
