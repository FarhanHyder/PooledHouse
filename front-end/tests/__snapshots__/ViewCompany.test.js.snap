// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewCompany component renders correct for multiple elements 1`] = `
<div
  className="container"
>
  <div
    className="position-left"
  >
    <h2
      className="card-title text-white bg-success rounded-lg"
    >
      Poolinos
    </h2>
    <div
      className="row"
    >
      <div
        className="col-6 btn-group"
      >
        <button
          className="btn btn-info"
          onClick={[Function]}
          type="button"
        >
          Description
        </button>
        <button
          className="btn btn-dark"
          onClick={[Function]}
          type="button"
        >
          Tips Information
        </button>
        <button
          className="btn btn-secondary"
          onClick={[Function]}
          type="button"
        >
          Tips History
        </button>
      </div>
      <h5
        className=" text-right col-6 text-success position-right"
      >
        Average Tips $
        12.89
      </h5>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="container card"
    >
      <div
        className="table-responsive-sm"
      >
        <table
          className="table table-dark table-striped table-bordered table-hover"
        >
          <thead
            className="bg-info"
          >
            <tr>
              <th>
                #
              </th>
              <th>
                Company Locations
              </th>
              <th>
                Average Reported Tips
              </th>
              <th>
                Tips in This Zip Code area
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                1
              </td>
              <td>
                292 Bowery, New York, NY 10012
              </td>
              <td>
                $
                16.36
                 / Hour
              </td>
              <td>
                $
                16.36
                 / Hour
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`ViewCompany component renders correct for single element 1`] = `
<div
  className="container"
>
  <div
    className="position-left"
  >
    <h2
      className="card-title text-white bg-success rounded-lg"
    >
      Poolinos
    </h2>
    <div
      className="row"
    >
      <div
        className="col-6 btn-group"
      >
        <button
          className="btn btn-info"
          onClick={[Function]}
          type="button"
        >
          Description
        </button>
        <button
          className="btn btn-dark"
          onClick={[Function]}
          type="button"
        >
          Tips Information
        </button>
        <button
          className="btn btn-secondary"
          onClick={[Function]}
          type="button"
        >
          Tips History
        </button>
      </div>
      <h5
        className=" text-right col-6 text-success position-right"
      >
        Average Tips $
        12.89
      </h5>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="container card"
    >
      <div
        className="table-responsive-sm"
      >
        <table
          className="table table-dark table-striped table-bordered table-hover"
        >
          <thead
            className="bg-info"
          >
            <tr>
              <th>
                #
              </th>
              <th>
                Company Locations
              </th>
              <th>
                Average Reported Tips
              </th>
              <th>
                Tips in This Zip Code area
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                1
              </td>
              <td>
                292 Bowery, New York, NY 10012
              </td>
              <td>
                $
                17.4
                 / Hour
              </td>
              <td>
                $
                17.4
                 / Hour
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
